### PR TITLE
Reclassify neighbors

### PR DESCRIPTION
**Pull request by pedrorponga**

### CODE REVIEW

File: skfda/misc/metrics.py
Comment (vnmabus) [inline skfda/misc/metrics.py:114]: These changes are in other PR.

File: skfda/ml/classification/centroid_classifiers.py
Comment (vnmabus) [inline skfda/ml/classification/centroid_classifiers.py:107]: This is also in other PR. Please split the changes and tell me the intended order of the PRs, as there are dependencies between them.

**Review by vnmabus — CHANGES_REQUESTED** [review]

File: skfda/ml/classification/centroid_classifiers.py
Comment (pedrorponga) [reply skfda/ml/classification/centroid_classifiers.py:107]: This branch was created based on the feature/DTM branch. I thought it was OK to do such things...

Comment (pedrorponga) [reply skfda/ml/classification/centroid_classifiers.py:107]: Maybe it's best to wait until feature/DTM can be merged to merge this branch

Comment (vnmabus) [reply skfda/ml/classification/centroid_classifiers.py:107]: It is ok, I think. I just need to be aware, in order to merge the other before.

File: skfda/ml/neighbors_outlier.py
Comment (vnmabus) [inline skfda/ml/neighbors_outlier.py:10]: The outlier detectors are not in `ml`, but in `exploratory/outliers`.

**Review by vnmabus — CHANGES_REQUESTED** [review]

File: skfda/ml/clustering/kmeans.py
Comment (pedrorponga) [reply skfda/ml/clustering/kmeans.py:15]: Sorry, I thought this had already been discussed. It appears as one of the remaining issues in the documentation. However, it does appear crossed out...

File: skfda/ml/classification/__init__.py
Comment (pedrorponga) [reply skfda/ml/classification/__init__.py:4]: Aren't these centroid-based classifiers?

File: skfda/ml/clustering/kmeans.py
Comment (vnmabus) [reply skfda/ml/clustering/kmeans.py:15]: Well, we did discuss it in the past, but I am not sure why we crossed it. I did not remember that we had left it written in an issue, sorry for assuming that you removed it without checking it with anyone. That said, I completely support the deletion.

File: skfda/ml/classification/__init__.py
Comment (vnmabus) [reply skfda/ml/classification/__init__.py:4]: Well, yes, but they are also distance based (although so are the neighbors ones). Maybe we should leave the modules private until we discuss a possible classification.

File: skfda/ml/classification/neighbors_classifiers.py
Comment (pedrorponga) [reply skfda/ml/classification/neighbors_classifiers.py:1]: True! Would you prefer neighbors_classifiers or neighbors? We could also change it in regression to neighbors_regression, and consequently maybe also linear to linear_regression... I added _classifier trying to be consistent with other file names in the classification folder

File: skfda/ml/classification/__init__.py
Comment (pedrorponga) [reply skfda/ml/classification/__init__.py:4]: Let's do that

File: setup.cfg
Comment (vnmabus) [inline setup.cfg:27]: Please order the ignored errors alphabetically.

File: setup.cfg
Comment (vnmabus) [inline setup.cfg:68]: Falta salto de línea.

File: setup.cfg
Comment (vnmabus) [inline setup.cfg:38]: Si pide solo docstring para el módulo, tampoco lo veo mal.

File: skfda/ml/classification/_centroid_classifiers.py
Comment (vnmabus) [reply skfda/ml/classification/_centroid_classifiers.py:173]: Deberíamos poner el strictness de darglint a "long" para que nos permita tener docstrings cortos sin especificar argumentos, return, etc, pero que compruebe los argumentos si los ponemos.

File: skfda/ml/classification/neighbors_classifiers.py
Comment (vnmabus) [reply skfda/ml/classification/neighbors_classifiers.py:280]: ¿Lo has resuelto?

**Review by vnmabus — CHANGES_REQUESTED** [review]

File: setup.cfg
Comment (pedrorponga) [reply setup.cfg:38]: Our __init__.py files currently do not have docstrings and they would add little information. But we could indeed add them.

File: skfda/ml/classification/_centroid_classifiers.py
Comment (pedrorponga) [reply skfda/ml/classification/_centroid_classifiers.py:173]: Yes. However, I think that in this case, it would be better to not have docstrings in `__init__` methods. Classes docstrings all ready document the parameters and moreover this is consistent with Google`s style guide.

Comment (vnmabus) [reply skfda/ml/classification/_centroid_classifiers.py:173]: I agree. It is consistent with scikit-learn too.

File: skfda/ml/classification/neighbors_classifiers.py
Comment (pedrorponga) [reply skfda/ml/classification/neighbors_classifiers.py:280]: No, I haven't found the option in isort yet

File: skfda/ml/classification/_centroid_classifiers.py
Comment (vnmabus) [reply skfda/ml/classification/_centroid_classifiers.py:173]: Whoops, I switched languages before 😅.

File: skfda/ml/classification/neighbors_classifiers.py
Comment (pedrorponga) [reply skfda/ml/classification/neighbors_classifiers.py:280]: Done

File: skfda/ml/classification/depth_classifiers.py
Comment (vnmabus) [inline skfda/ml/classification/depth_classifiers.py:1]:
```suggestion
"""Depth-based models for supervised classification."""
```

File: skfda/ml/clustering/kmeans.py
Comment (vnmabus) [inline skfda/ml/clustering/kmeans.py:702]: I usually do not leave spaces around the exponential operator, to easily see the terms in a sum or product. Do our recent linter complain about that?

File: skfda/ml/regression/neighbors.py
Comment (vnmabus) [inline skfda/ml/regression/neighbors.py:333]: Start the sentence on the next line.

File: tests/test_neighbors.py
Comment (vnmabus) [inline tests/test_neighbors.py:95]:
```suggestion
        """Test scalar regression, predicts mode location."""
```

**Review by vnmabus — CHANGES_REQUESTED** [review]

File: skfda/ml/clustering/kmeans.py
Comment (pedrorponga) [reply skfda/ml/clustering/kmeans.py:702]: It is okay do to that.

File: skfda/ml/regression/neighbors.py
Comment (pedrorponga) [reply skfda/ml/regression/neighbors.py:333]: Not sure what you're asking.

Comment (vnmabus) [reply skfda/ml/regression/neighbors.py:333]: This
```suggestion
        """
        Return distances and neighbors of given sample.
```

Comment (pedrorponga) [reply skfda/ml/regression/neighbors.py:333]: Why do we need to do that?

Comment (pedrorponga) [reply skfda/ml/regression/neighbors.py:333]: https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html

Comment (vnmabus) [reply skfda/ml/regression/neighbors.py:333]: I usually prefer that all lines start on the same column, but it is only a preference.

File: setup.cfg
Comment (pedrorponga) [reply setup.cfg]: Actually, we do have some with docstrings. This needs to change.

File: skfda/ml/regression/neighbors.py
Comment (pedrorponga) [reply skfda/ml/regression/neighbors.py:333]: Okey, as it seems more standard to keep it in the same line and apparently that use is more spread in our code, I wouldn't change this.

Comment (vnmabus) [reply skfda/ml/regression/neighbors.py:333]: Ok.

File: skfda/ml/clustering/kmeans.py
Comment (vnmabus) [reply skfda/ml/clustering/kmeans.py]: So do it, then.

Comment (pedrorponga) [reply skfda/ml/clustering/kmeans.py]: It should already be changed.

Comment (vnmabus) [reply skfda/ml/clustering/kmeans.py]: Ok, I see it now

**Review by vnmabus — DISMISSED** [review]

File: skfda/ml/clustering/__init__.py
Comment (vnmabus) [reply skfda/ml/clustering/__init__.py:1]: Maybe not the final commit after all! 😂😂😂

Comment (pedrorponga) [reply skfda/ml/clustering/__init__.py:1]: 😒

**Review by vnmabus — APPROVED** [review]

**Pull request merged**
